open-craft/opencraft
# BB-1919: Add registration flow

This PR introduces the new Ocim registration pages with the basic registration flow.

**Testing instructions:**
1. `npm i && npm start`
2. Go to http://localhost:3000/registration/domain.
3. Insert `existing` and click "Check availability".
4. Check that an error message is displayed.
5. Write something and check that the error message immediately vanishes after editing the field.
6. Test that that registration form is working (next/back buttons work, data is store on the state).
7. Check that the styles are similar to the ones in the mockups.

Since the validators aren't implemented yet, change a few values in the `registrationReducer` to show when fields have feedback error messages. Example: show error on password field:
```
    case RegistrationActions.Types.ROOT_STATE_UPDATE:
      // Merge state without erasing previous values
      return {
        loading: state.loading,
        registrationData: {
          ...state.registrationData,
          ...action.data.registrationData
        },
        registrationFeedback: {
          ...state.registrationFeedback,
          ...action.data.registrationFeedback,
          // ADD THIS LINE
          password: "This password already exists!"
        }
      };
```

**Reviewer:**
